crates/ty_python_semantic/resources/mdtest/assignment/annotations.md

```diff
@@ -324,27 +324,21 @@ class X[T]:
     def __init__(self, value: T):
         self.value = value
 
-a: X[int] = X(1)
-reveal_type(a)  # revealed: X[int]
+x1: X[int] = X(1)
+reveal_type(x1)  # revealed: X[int]
 
-b: X[int | None] = X(1)
-reveal_type(b)  # revealed: X[int | None]
+x2: X[int | None] = X(1)
+reveal_type(x2)  # revealed: X[int | None]
 
-c: X[int | None] | None = X(1)
-reveal_type(c)  # revealed: X[int | None]
+x3: X[int | None] | None = X(1)
+reveal_type(x3)  # revealed: X[int | None]
 
-def _[T](a: X[T]):
-    b: X[T | int] = X(a.value)
-    reveal_type(b)  # revealed: X[T@_ | int]
+def _[T](x1: X[T]):
+    x2: X[T | int] = X(x1.value)
+    reveal_type(x2)  # revealed: X[T@_ | int]
 
-d: X[Any] = X(1)
-reveal_type(d)  # revealed: X[Any]
-
-def _(flag: bool):
-    # TODO: Handle unions correctly.
-    # error: [invalid-assignment] "Object of type `X[int]` is not assignable to `X[int | None]`"
-    a: X[int | None] = X(1) if flag else X(2)
-    reveal_type(a)  # revealed: X[int | None]
```

Comment:
This is actually a more general issue (see https://github.com/astral-sh/ty/issues/1543), so it doesn't make sense to test it here.